backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,532 @@
+    AOTITensorHandle tensor,
+    int64_t* ret_size) {
+    throw std::runtime_error("Cannot get storage size on ETensor");
+  }
+  AOTITorchError aoti_torch_create_tensor_from_blob_v2(
+    void* data,
+    int64_t ndim,
+    const int64_t* sizes_ptr,
+    const int64_t* strides_ptr,
+    int64_t storage_offset,
+    int32_t dtype,
+    int32_t device_type,
+    int32_t device_index,
+    AOTITensorHandle* ret_new_tensor,
+    int32_t layout,
+    const uint8_t* opaque_metadata,
+    int64_t opaque_metadata_size) {
+    throw std::runtime_error("Not creating Tensor from blob here");
+  }
+  AOTITorchError aoti_torch_create_cuda_stream_guard(
+    void* stream,
+    int32_t device_index,
+    CUDAStreamGuardHandle* ret_guard) {
+    std::cout<<"Entering stream guard for device " << device_index << std::endl;
+    return Error::Ok;
+  }
+  AOTITorchError aoti_torch_delete_cuda_stream_guard(
+    CUDAStreamGuardHandle guard) {
+    std::cout<<"Exiting stream guard" << std::endl;
+    return Error::Ok;
+  }
+  int aoti_torch_device_type_cpu() {
+    // Let's say cpu is 0 for ET as well
+    return 0;
+  }
+  __attribute__((__visibility__("default"))) int32_t aoti_torch_device_type_cuda() {
+    // Let's say cuda is 1 for ET as well
+    return 1;
+  }
+  __attribute__((__visibility__("default"))) int32_t aoti_torch_dtype_float32() {
+    // Let assume the dtype here is all we will support
+    return 6;
+  }
+  AOTITorchError aoti_torch_delete_tensor_object(AOTITensorHandle tensor) {
+    std::cout << "Deleting " << tensor << std::endl;
+    for (auto it = tensors.begin(); it != tensors.end(); ++it) {
+        if (it->get() == tensor) {
+            tensors.erase(it);
+            break; // Exit the loop once the element is found and removed
+        }
+    }
+    return Error::Ok;
+  }
+  AOTITorchError aoti_torch_create_tensor_from_blob(
+    void* data,
+    int64_t ndim,
+    const int64_t* sizes_ptr,
+    const int64_t* strides_ptr,
+    int64_t storage_offset,
+    int32_t dtype,
+    int32_t device_type,
+    int32_t device_index,
+    AOTITensorHandle* ret_new_tensor) {
+    throw std::runtime_error("Should never create from blob");
+    return Error::NotSupported;
+  }
+  AOTITorchError aoti_torch_empty_strided(
+    int64_t ndim,
+    const int64_t* sizes_ptr,
+    const int64_t* strides_ptr,
+    int32_t dtype,
+    int32_t device_type,
+    int32_t device_index,
+    AOTITensorHandle* ret_new_tensor) {
+      // This requires us to reserve CUDA memory and put it into a ETensor
+      void* ptr;
+      int64_t numel = 1;
+      for (int i = 0; i < ndim; i++) {
+        numel *= sizes_ptr[i];
+      }
+
+      if (dtype != 6) { // throw if not float32
+        throw std::runtime_error("Need to implement empty_strided for non-float32");
+      }
+
+      int64_t nbytes = numel * 4;
+
+      if (device_type == 1) { // cuda
+        std::cout << "Allocating " << nbytes << " bytes on CUDA " << std::endl;
```

Comment:
if you move this stuff to ET_LOG at the verbose log level you might not even have to delete it to commit